vscode-antimony/src/extension.ts

```diff
@@ -185,6 +225,78 @@ export async function activate(context: vscode.ExtensionContext) {
 			triggerUpdateDecorations(true);
 		}
 	}, null, context.subscriptions);
+
+	const sbmlFileNameToPath = new Map();
+
+	// when user opens XML
+	if (roundTripping) {
+		vscode.workspace.onDidOpenTextDocument(async event => {
+			triggerSBMLEditor(event, sbmlFileNameToPath);
+		});
+	
+		vscode.workspace.onDidCloseTextDocument((closedDoc) => {
+			const fileName = path.basename(closedDoc.fileName, '.git');
+			const pathName = path.dirname(closedDoc.fileName);
+			const fullPath = path.join(pathName, fileName);
+			const pattern = /^(.+?).ant/;
+			if (pattern.test(fileName) && pathName === os.tmpdir()) {
+				vscode.window.showInformationMessage('Do you want to save the changes to the original SBML file?\n Warning: edits will be lost if not saved!', 'Save', 'Discard')
+				.then((choice) => {
+					if (choice === 'Save') {
+						vscode.workspace.openTextDocument(fullPath).then((doc) => {
+							vscode.commands.executeCommand('antimony.antStrToSBMLStr', doc.getText())

```

Comment:
Consider putting this in a method so you don't have to regurgitate the same code in two places. From line 247-260 and 277-288. Since you use vscode.window.showinfomessage, you can pass a parameter and an if statement to call this if its param = ondidsave and not if param = ondidclose